### PR TITLE
feat: 🍰 Implement `UpdateGroup` Resolver

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pull Request

Implement `UpdateGroup` resolver.

### Issues

- relates #5131
- relates #5059

### Todo

- [ ] GraphQL
  - [ ] add group settings ( details see https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/5059#user-content-group-settings )
    - [x] categories
    - [x] avatar
    - [ ] location
- [ ] tests:
  - [x] general
  - [x] slugification
    - [x] if we change the name so that an existing slug will be generated that it is made unique by `*-1`
  - [x] test if resolver `Group` finds group by `id`
  - [ ] does avatar testing make sense this way in `backend/src/schema/resolvers/user_management.spec.js`
    - [ ] minimum: await is missing …
    - [ ] search for other appearances of `avatar: Factory.build('image',` or `Factory.build('image'` without await
    - [x] created an issue #5256 
- [x] cleanup: search for `Wolle`

### CODE REVIEW

File: backend/src/schema/resolvers/groups.js [hunk 1]
```diff
@@ -131,6 +133,76 @@ export default {
         session.close()
       }
     },
+    UpdateGroup: async (_parent, params, context, _resolveInfo) => {
+      const { categoryIds } = params
+      const { id: groupId, avatar: avatarInput } = params
+      delete params.categoryIds
+      if (CONFIG.CATEGORIES_ACTIVE && categoryIds) {
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.js:140]: What happens when categories are active but `categoriesIds` is `undefined`

File: backend/src/schema/resolvers/groups.spec.js [hunk 2]
```diff
@@ -2043,5 +2140,211 @@ describe('in mode', () => {
...
+              authenticatedUser = await user.toJson()
+            })
+
+            it('has set the settings', async () => {
+              await expect(
+                mutate({
+                  mutation: updateGroupMutation,
+                  variables: {
+                    id: 'my-group',
+                    name: 'The New Group For Our Coutry',
```
Comment (Mogge) [inline backend/src/schema/resolvers/groups.spec.js:2221]: Mind the typo?

**Review by Mogge — CHANGES_REQUESTED** [review]
Looks very good! Some question ...

File: backend/src/schema/resolvers/groups.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.js:140]: If `categoriesIds` are `null` or `undefined` they are not changed in any way.

If `categoriesIds` are `[]` an error is thrown if `CONFIG.CATEGORIES_ACTIVE === true`.
If the array is not empty the old are deleted and new are set.

But there's no type check if it's an array. You thing it's needed?

Comment (Tirokk) [reply backend/src/schema/resolvers/groups.js:140]: The type check is done by GQL … 😉

File: backend/src/schema/resolvers/groups.spec.js (hunk 2 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/groups.spec.js]: Nice catch! 🍀
done

**Review by Mogge — APPROVED** [review]

**Pull request merged**